qkernel/src/lib.rs

```diff
@@ -483,20 +517,47 @@ cfg_if::cfg_if! {
             vdsoParamAddr: u64,
             vcpuCnt: u64,
             autoStart: bool,
+            vm_type: u64
```

Comment:
I have a feeling this will cause too much runtime dependency (also the case for `IS_SEV_SNP` and `ENABLE_CC` etc). In other words you will have check these flags quite frequently. I don't know what's an alternative though, maybe we can discuss this at some point.